presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp

```diff
@@ -318,17 +320,78 @@ TEST_F(UnsafeRowShuffleTest, endToEnd) {
   }
 }
 
+TEST_F(UnsafeRowShuffleTest, StorageBasedShuffle) {
+  size_t numPartitions = 5;
+  // Registering new shuffle related operators.
+  exec::Operator::registerOperator(
+      std::make_unique<PartitionAndSerializeTranslator>());
+  exec::Operator::registerOperator(std::make_unique<ShuffleWriteTranslator>());
+
+  // Creating a local file system storage based shuffle.
+  velox::filesystems::registerLocalFileSystem();
+  auto rootPath = velox::exec::test::TempDirectoryPath::create()->path;
+  std::filesystem::create_directory(rootPath);
```

Comment:
TempDirectoryPath::create() creates a directory. It deletes it on destruction. Instead of creating a directory, deleting it, then creating it again, consider creating it and not deleting. This way we'll also ensure that all the files get deletes on test completion. Otherwise, we'll be leaving lots of files behind.

```
auto rootDirectory = velox::exec::test::TempDirectoryPath::create();
auto rootPath = rootDirectory->path;
```